perfeed/tools/weekly_summarizer.py

```diff
@@ -0,0 +1,98 @@
+import asyncio
+import time
+from datetime import datetime, timedelta
+
+from jinja2 import Environment, StrictUndefined
+
+from perfeed.config_loader import settings
+from perfeed.git_providers.base import BaseGitProvider
+from perfeed.git_providers.github import GithubProvider
+from perfeed.llms.base_client import BaseClient
+from perfeed.llms.ollama_client import OllamaClient
+from perfeed.models.pr_summary import PRSummary
+from perfeed.tools.pr_summarizer import PRSummarizer
+from perfeed.utils.utils import json_output_curator
+
+
+# python perfeed.py -users jimmytai chihangwang -repo perfeed-backend -period 1w
+class WeeklySummarizer:
```

Comment:
can we isolate a function that fetch and store PRSummary at given time? so that it can be reused somewhere else like for gathering all PRs for week over week. And let WeeklySummarizer does one thing - load and invoke model

Reply:
If I understand correctly, you mean `WeeklySummarizer` should only fetch `PRSummary` from DB and pass to LLM? As for fetching and storing `PRSummary` in DB, it's done by something else that's reusable. I think we can do that, but I'd like to separate that out later and have the main backbone checked in first so we can all work on it.

Reply:
To have a `fetchPRsummary` function that loads from DB if available. If not, loop and fetch via `PRSummerizer`. This can be reused in many scenarios.

Reply:
Yeah, let me add a todo for that as in `load the PR summaries from the pervious batch if exists.` I'd like to keep this PR smaller